contracts/extensions/bridge-operator-governance/BridgeManagerCallbackRegister.sol

```diff
@@ -0,0 +1,123 @@
+   */
+  function registerCallbacks(address[] calldata registers) external onlySelfCall returns (bool[] memory registereds) {
+    registereds = _registerCallbacks(registers);
+  }
+
+  /**
+   * @inheritdoc IBridgeManagerCallbackRegister
+   */
+  function unregisterCallbacks(
+    address[] calldata registers
+  ) external onlySelfCall returns (bool[] memory unregistereds) {
+    unregistereds = _unregisterCallbacks(registers);
+  }
+
+  /**
+   * @dev Internal function to register multiple callbacks with the bridge.
+   * @param registers The array of callback addresses to register.
+   * @return registereds An array indicating the success status of each registration.
+   */
+  function _registerCallbacks(
+    address[] memory registers
+  ) internal nonDuplicate(registers) returns (bool[] memory registereds) {
+    uint256 length = registers.length;
+    registereds = new bool[](length);
+    EnumerableSet.AddressSet storage _callbackRegisters = _getCallbackRegisters();
+    address register;
+    bytes4 callbackInterface = type(IBridgeManagerCallback).interfaceId;
+    for (uint256 i; i < length; ) {
+      register = registers[i];
+
+      _requireHasCode(register);
+      _requireSupportsInterface(register, callbackInterface);
+      registereds[i] = _callbackRegisters.add(register);
+
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /**
+   * @dev Internal function to unregister multiple callbacks from the bridge.
+   * @param registers The array of callback addresses to unregister.
+   * @return unregistereds An array indicating the success status of each unregistration.
+   */
+  function _unregisterCallbacks(
+    address[] memory registers
+  ) internal nonDuplicate(registers) returns (bool[] memory unregistereds) {
+    uint256 length = registers.length;
+    unregistereds = new bool[](length);
+    EnumerableSet.AddressSet storage _callbackRegisters = _getCallbackRegisters();
+
+    for (uint256 i; i < length; ) {
+      unregistereds[i] = _callbackRegisters.remove(registers[i]);
+
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /**
+   * @dev Internal function to notify all registered callbacks with the provided function signature and data.
+   * @param callbackFnSig The function signature of the callback method.
+   * @param inputs The data to pass to the callback method.
+   */
+  function _notifyRegisters(bytes4 callbackFnSig, bytes memory inputs) internal {
+    address[] memory registers = _getCallbackRegisters().values();
+    uint256 length = registers.length;
+    if (length != 0) {
+      bool[] memory statuses = new bool[](length);
+      bytes[] memory returnDatas = new bytes[](length);
+      bytes memory callData = abi.encodePacked(callbackFnSig, inputs);
+
+      for (uint256 i; i < length; ) {
+        (statuses[i], returnDatas[i]) = registers[i].call(callData);
+
+        unchecked {
+          ++i;
+        }
+      }
+
+      emit Notified(registers, statuses, returnDatas);
```

Comment:
Consider including your calldata(s) to support debugging purposes

Reply:
Because the calldata is deterministic per transaction so not including them is also fine

Reply:
But when you have any intention to notify many types in a single transaction, please add the calldata to the event

Reply:
updated.